sd_mecha/merge_methods.py

```diff
@@ -130,29 +130,51 @@ def add_cosine_generic(a: Tensor, b: Tensor, alpha: float, similarity: Tensor) -
     return weighted_sum.__wrapped__(a, b, alpha=k)
 
 
+"""
+### ties_sum ###
+- `delta`: $$ \hat{\tau}_t $$
+- `signs`: $$ \gamma_t $$ 
+- `final_sign`: $$ \gamma_m^p = sgn(\sum_{t=1}^n \hat{\tau}_t^p) $$ 
+- `delta_filters`: $$ \{ \gamma_t^p = \gamma_m^p \} $$
+- `param_counts`: $$ |A^p| $$
+- `filtered_delta`: $$ \sum_{t\in{A^p}} \hat{\tau}_t^p $$
+- `return`: $$ \lambda * \tau_m $$
+"""
 @convert_to_recipe
 def ties_sum(  # aka add_difference_ties
     *models: Tensor | LiftFlag[MergeSpace.DELTA],
     alpha: Hyper,
     k: Hyper = 0.2,
     **kwargs,
 ) -> Tensor | LiftFlag[MergeSpace.DELTA]:
-    deltas = []
-    signs = []
-    for m in models:
-        deltas.append(filter_top_k(m, k))
-        signs.append(torch.sign(deltas[-1]))
-
-    signs = torch.stack(signs, dim=0)
-    final_sign = torch.sign(torch.sum(signs, dim=0))
+    
+    # Step 1: Trim redundant parameters
+
+    # $$ \hat{\tau}_t $$ O(N) in space
+    deltas = [filter_top_k(m, k) for m in models]
     deltas = torch.stack(deltas, dim=0)
+
+    # $$ \gamma_t $$ 
+    signs = torch.sign(deltas)
+
+    # $$ \gamma_m^p = sgn(\sum_{t=1}^n \hat{\tau}_t^p) $$
+    final_sign = torch.sign(torch.sum(deltas, dim=0)) 
+
+    # Step 3: Disjoint merge.
+
+    # $$ \{ \gamma_t^p = \gamma_m^p \} $$
     delta_filters = (signs == final_sign).float()
-    filtered_delta = (deltas * delta_filters).sum(dim=0)
 
+    # $$ |A^p| $$
     param_counts = torch.sum(delta_filters, dim=0)
-    return alpha * torch.nan_to_num(filtered_delta / param_counts)
 
+    # $$ \sum_{t\in{A^P}} \hat{\tau}_t^p $$
+    filtered_delta = (deltas * delta_filters).sum(dim=0)
+
+    # $$ \lambda * \tau_m $$
+    return alpha * torch.nan_to_num(filtered_delta / param_counts)
 
+# $$ keep_topk_reset_rest_to_zero(\tau_t, k) $$
```

Comment:
while I think it's a good idea to add comments referring to ties code inside the ties function, `filter_top_k` is used by other methods too so it might be confusing or there could be semantic conflicts if we want to add pseudo algo comments for all papers.

Reply:
> pushing to this branch
Which branch? Sure you can, tell me if you need actions in my fork. This PR is already good enough for me.

> used by other methods too
Oh this is not expected. Just specify it as "in TIES", and use other notation for other algorithms. Function code is untouched.

Reply:
what about we annotate the *call* of the function instead of the definition? something like this
```python
deltas = [
    # $$ keep_topk_reset_rest_to_zero(\tau_t, k) $$
    filter_top_k(m, k)
    for m in models
]
```